lib/statsig/configs.ex

```diff
@@ -0,0 +1,104 @@
+defmodule Statsig.Configs do
+  use GenServer
+  require Logger
+
+  @table_name :statsig_configs
+
+  defmodule State do
+    @type t :: %__MODULE__{
+      last_sync_time: integer(),
+      reload_timer: reference() | nil,
+      reload_interval: integer()
+    }
+
+    defstruct last_sync_time: 0,
+              reload_timer: nil,
+              reload_interval: 10_000
+
+    def new(opts \\ []) do
+      reload_interval = Keyword.get(opts, :reload_interval, 10_000)
+      %__MODULE__{reload_interval: reload_interval}
+    end
+
+    def schedule_reload(%__MODULE__{} = state) do
+      timer = Process.send_after(self(), :reload_configs, state.reload_interval)
+      %__MODULE__{state | reload_timer: timer}
+    end
+  end
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, State.new(), name: __MODULE__)
+  end
+
+  def lookup(name, type) do
+    :ets.lookup(@table_name, {name, type})
+  end
+
+  @impl true
+  def init(_) do
+    :ets.new(@table_name, [:named_table, :set, :public])
+    state = State.new(reload_interval: reload_interval())
+    |> State.schedule_reload()
```

Comment:
nit: commit to the pipeline
```suggestion
    state = [reload_interval: reload_interval()] 
    |> State.new()
    |> State.schedule_reload()
```